pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -291,24 +336,75 @@ func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
 	return nil
 }
 
-func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig, client mdbClient.Client) (automationconfig.AutomationConfig, error) {
-	domain := getDomain(mdb.ServiceName(), mdb.Namespace, "")
+// checkTLSConfig will check that the configured ConfigMap and Secret exist and that they have the correct fields.
+func (r *ReplicaSetReconciler) checkTLSConfig(mdb mdbv1.MongoDB) (bool, error) {
+	if !mdb.Spec.TLS.Enabled {
+		return false, nil
+	}
 
-	currentAc, err := getCurrentAutomationConfig(client, mdb)
-	if err != nil {
-		return automationconfig.AutomationConfig{}, err
+	r.log.Info("Ensuring TLS is correctly configured")
+
+	// Ensure CA ConfigMap exists
+	var caConfigMap corev1.ConfigMap
+	configMapName := types.NamespacedName{Name: mdb.Spec.TLS.CAConfigMapRef, Namespace: mdb.Namespace}
+	if err := r.client.Get(context.TODO(), configMapName, &caConfigMap); err != nil {
+		return errors.IsNotFound(err), err
```

Comment:
if you want to use IsNotFound you have to check and then either return true, nil or false, err

you should not return a value and an error

Reply:
@fabianlindfors I think the confusion comes from this: don't spare an if and return an error if it's not an IsNotFound otherwise return a retry true without error

Reply:
Check out the discussion we had in the other thread. I've implemented a solution which is more idiomatic.

Reply:
@fabianlindfors you mention you need a 3 case solution: bool, err allows for 3 cases as shown above
this is making it un-necessarily complicated
why bool, err is not working for you? you only need the struct to hold the log message: logging it right away would solve this and also acting closest to the error is a golang best practice (as here our action is only to log, why not doing it here)

Reply:
The only reason is trying to stay consistent with our style of bubbling errors up to the `Reconcile()` and then logging there. Adding complexity just to stay in line with that though doesn't feel right, I'll change it!

Reply:
@LouisPlisso This has now been changed. Please have another look!